pkg/csr/csr.go

```diff
@@ -0,0 +1,260 @@
+/*
+Based on https://github.com/openshift/cluster-machine-approver/tree/master/pkg/controller
+Cluster machine approver approves CSR's from machines, hence we cannot use the code from
+the package for approving CSR's from BYOH instances which may not have reference to a
+machine object
+*/
+
+package csr
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	certificates "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+)
+
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=certificatesigningrequests/approval,verbs=update
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=certificatesigningrequests,verbs=get;list;watch
+//+kubebuilder:rbac:groups="certificates.k8s.io",resources=signers,verbs=Approve,resourceNames=kubernetes.io/kube-apiserver-client-kubelet;kubernetes.io/kubelet-serving
+
+// kubeletClientUsages contains the permitted key usages from a kube-apiserver-client-kubelet signer
+var kubeletClientUsages = []certificates.KeyUsage{
+	certificates.UsageKeyEncipherment,
+	certificates.UsageDigitalSignature,
+	certificates.UsageClientAuth,
+}
+
+// kubeletServerUsages contains the permitted key usages from a kubelet-serving signer
+var kubeletServerUsages = []certificates.KeyUsage{
```

Comment:
This is a slice that can be mutated. 

Does `kubeletServerUsages` meant to be a constant? If so, consider something like:
```
func getKubeletServerUsages() []certificates.KeyUsage {
        // where you return a copy of a fixed array
	return [...]certificates.KeyUsage{    }
}
```

Reply:
Do we really need to do this given this is a private member that is not exposed to the outside world?

Reply:
Does this need to be a global variable though?